demos-europe/demosplan-core
# bug(refs: T29346): fix map modal is not toggling

https://yaits.demos-deutschland.de/T29346


### Description
- In AssessmentTableCard the reference to the modal was not correct. To make this work it would be needed to reference the modal way deeper (this.$parent.$parent.$parent.$parent... would be the correct reference). However this seemes overly complicated to follow if we need to change this component anytime in the future.

- Hence an approach by emitting a custom event which in the end toggles the modal was a better solution from my perspective.

- DpMapModal listens for the event toggleModal which will actually toggle the modal and the event gets fired in DpAssessmentTableCard

### How to review/test

- login as fpa
- navigate to a-table
- click the marked button (screenshot)
- map modal should open

### PR Checklist


Delete the checkbox if it doesn't apply/isn't necessary.


![Screenshot 2023-02-14 at 18-58-47 ⚓ T29346 AT Verortete STN Einzeichnung - Ansehen funktioniert nicht](https://user-images.githubusercontent.com/91727189/218819826-948f80db-bed5-40dc-9283-c4f13bfcb7de.png)
